noria-server/src/controller/inner.rs

```diff
@@ -721,30 +714,12 @@ impl ControllerInner {
             .collect()
     }
 
-    fn find_view_for(&self, node: NodeIndex) -> Option<NodeIndex> {
-        // reader should be a child of the given node. however, due to sharding, it may not be an
-        // *immediate* child. furthermore, once we go beyond depth 1, we may accidentally hit an
-        // *unrelated* reader node. to account for this, readers keep track of what node they are
-        // "for", and we simply search for the appropriate reader by that metric. since we know
-        // that the reader must be relatively close, a BFS search is the way to go.
-        let mut bfs = Bfs::new(&self.ingredients, node);
-        let mut reader = None;
-        while let Some(child) = bfs.next(&self.ingredients) {
-            if self.ingredients[child]
-                .with_reader(|r| r.is_for() == node)
-                .unwrap_or(false)
-            {
-                reader = Some(child);
-                break;
-            }
-        }
-
-        reader
-    }
-
     /// Obtain a `ViewBuilder` that can be sent to a client and then used to query a given
-    /// (already maintained) reader node called `name`.
-    pub fn view_builder(&self, name: &str) -> Option<ViewBuilder> {
+    /// (already maintained) reader node. If the view has replicas, a `ViewBuilder` is returned
+    /// for each view in round robin order.
```

Comment:
for each _replica_? I wonder if there's a way we can clear up the language here and elsewhere... In particular, it seems like sometimes refer to "replicas" as all the _other_ `Reader`s except the "first one", but we also store all the `Reader`s in a single `Vec` called `replicas`. Which is it? Maybe the way way to talk about this is that a "view" has one or more "instances", and the instances are all replicas of one another. "Instance" then maps directly onto `Reader`. Sadly we use the word "instance" in all sorts of settings, so we may need something less common. "Manifestation" perhaps? It could be that the word we actually want to use here is "materialization" (i.e., a given view may have multiple (identical) materializations). @ms705 ?